episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,871 @@
+case nuclei).
+
+A simple approach to create a mask is to use a method called 'thresholding'. 
+Thresholding works best when there is a clear separation in pixel values between 
+the class of interest and the rest of the image. In these cases, thresholding 
+involves choosing one or more specific pixel values to use as the cut-off 
+between background and your class of interest. For example, in this image, you 
+could choose a pixel value of 100 and set all pixels with a value less than that 
+as background, and all pixels with a value greater than that as nuclei. How do 
+we go about choosing a good threshold though?
+
+One common approach to choosing a threshold is to use the image histogram - 
+recall that we looked at histograms in detail in the [image display episode
+](image-display.md). Let's go ahead and open a histogram with:  
+`Plugins > napari Matplotlib > Histogram`
+
+![](fig/nuclei-histogram.png){alt="A histogram of the 29th z slice of Napari's 
+cell sample image"}
+
+From this histogram, we can see two peaks - a larger one at low intensity, then 
+a smaller one at higher intensity. Looking at the image, it would make sense if 
+the lower intensity peak represented the background, with the higher one 
+representing the nuclei. We can verify this by adjusting the contrast limits in 
+the layer controls. If we move the left contrast limits node to the right of the 
+low intensity peak (around 8266), we can still clearly see the nuclei:
+
+![](fig/contrast-limit-8266-nuclei.png){alt="Left, nuclei with gray colormap. 
+Right, histogram of the same image. Both with left contrast limit set 
+to 8266."}
+
+If we move the left contrast limits node to the right of the higher intensity 
+peak (around 28263), most of the nuclei disappear from the image:
+
+![](fig/contrast-limit-28263-nuclei.png){alt="Left, nuclei with gray colormap. 
+Right, histogram of the same image. Both with left contrast limit set 
+to 28263."}
+
+Recall that you can set specific values for the contrast limits by right 
+clicking on the contrast limits slider in Napari.
+
+This demonstrates that the higher intensity peak is the one we are interested 
+in, and we should set our threshold accordingly.
+
+We can set this threshold with some commands in Napari's console:
+
+```python
+
+# Get the image data for the nuclei
+nuclei = viewer.layers["nuclei"].data
+
+# Create mask with a threshold of 8266
+mask = nuclei > 8266
+
+# Add mask to the Napari viewer
+viewer.add_labels(mask)
+
+```
+
+You should see a mask appear that highlights the nuclei in brown. If we set the 
+nuclei contrast limits back to normal (select 'nuclei' in the layer list, then 
+drag the left contrast limits node back to zero), then toggle on/off the mask 
+or nuclei layers with the ![](
+https://raw.githubusercontent.com/napari/napari/main/napari/resources/icons/visibility.svg
+){alt="A screenshot of Napari's eye button" height='30px'} icon, you should see 
```

Comment:
Thinking back to the talk on digital accessibility, what's the right alt text to use in this situation? If you were using a screen reader it would read better if the alt. text was just "Napari's eye". The "A screenshot ... " is more suitable for the main figures. Maybe create a new issue and review these for all lessons after some research.

Reply:
I made a new issue for this here: https://github.com/HealthBioscienceIDEAS/microscopy-novice/issues/48